doc/developers/develop.rst

```diff
@@ -475,9 +475,16 @@ on a classifier, but not otherwise. Similarly, scorers for average precision
 that take a continuous prediction need to call ``decision_function`` for classifiers,
 but ``predict`` for regressors. This distinction between classifiers and regressors
 is implemented using the ``_estimator_type`` attribute, which takes a string value.
-It should be ``"classifier"`` for classifiers and ``"regressor"`` for
-regressors and ``"clusterer"`` for clustering methods, to work as expected.
-Inheriting from ``ClassifierMixin``, ``RegressorMixin`` or ``ClusterMixin``
+This attribute should have the following values to work as expected:
+
+- ``"classifier"`` for classifiers
+- ``"regressor"`` for regressors
+- ``"clusterer"`` for clustering methods
+- ``"outlier_detector"`` for outlier detectors
+- ``"DensityEstimator"`` for density estimators
```

Comment:
I agree that this matches our current code base; however, I find the casing style of the last tag value inconsistent with the others. However, I don't see any easy way to fix it without breaking backward compat because it's technically part of our public API.

I guess we have to live with it for now but keep it in mind if we want to release scikit-learn 2.0 one day.

/cc @adrinjalali as developer API specialist.

Reply:
in a rework of this document, I'd like to remove all of these and only document developing estimators via inheriting from our mixins. That would remove these parts anyway.

Otherwise. also gonna work on a PR to move this from the class attribute to a tag. I think it's doable, but need to actually implement it to be sure.

Reply:
> /cc @adrinjalali as developer API specialist.

I like the fancy title 😅